railties/lib/rails/generators/rails/app/templates/config/puma.rb

```diff
@@ -0,0 +1,9 @@
+threads_count = ENV.fetch("MAX_THREADS") { 5 }.to_i
```

Comment:
> that's really not how we configure Rails.

We do __all__ of our rails configuration behind environment variables:

```
$ RAILS_ENV=production rails server
```

For the puma config, we want something that can be over-rideable. We need to have `rails server` work out of the box with some sane defaults but be extensible. I don't know if there's another way to accomplish this. I don't think we should tie the interface of `rails server` flags to the implementation of Puma's server, `$ rails server --minimum-puma-threads=5 --maximum-puma-threads=5`. We could put this in a YML somewhere and have values for different environments but honestly I would set all 3 to the same values. Then that adds yet another place to look for configuration. We can't use Rail's config methods already in place because this code will get booted before your Rails code will.

In my perfect world we would figure out a way to better integrate AR's pool setting with a server's thread count setting. I.e. so if you increase your threads on your server then AR knows and uses a higher pool value. Right now you have to know about EVERYWHERE that uses a DB connection. So instead of 1 change you have to modify sidekiq config + puma config + AR pool size.